plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,648 @@
+Currently, the MEP is set to 7 days, so each period has a length of 3.5 days.
+
+
+#### Starting the Exit
+
+For simplicity, we block deposit transactions from using the MoreVP exit protocol.
+
+Any user may initiate an exit by presenting a spend transaction and proving that the transaction is exitable.
+The user must submit a bond, `exit bond`, for starting this action.
+This bond is later used to cover the cost for other users to publish statements about the canonicity of the transaction in question.
+
+We provide several possible mechanisms that allow a user to prove a transaction is exitable.
+First, remember that deposit transactions are exitable by default.
+Two ways in which spend transactions can be proven exitable are as follows:
+1. The user may present the transaction along with each of the transactions that created the inputs to the transaction, a Merkle proof of inclusion for each of these transactions, and a signature from each input owner.
+The contract can then validate that these transactions are the correct ones, that they were included in the chain, that the signatures are correct, and that the exiting transaction is correctly formed.
+This proves the exitability of the transaction.
+2. The user may present the transaction along signatures the user claims to be valid.
+The contract can validate that the exiting transaction is correctly formed.
+Another user can challenge one of these signatures by presenting some transaction that created an input such that the true input owner did not sign the signature.
+In this case, the exit would be blocked entirely and the challenging user would receive `exit bond`.
+
+Option (1) checks that a transaction is exitable when the exit is started.
+This has lower communication cost and complexity but higher up-front gas cost.
+This option also ensures that only a single exit on any given transaction can exist at any point in time.
+Option (2) allows a user to assert that a transaction is exitable, but leaves the proof to a challenge-response game.
+This is cheaper up-front but adds complexity.
+This option must permit multiple exits on the same transaction, as some exits may provide invalid signatures.
+
+These are not the only possible mechanisms that prove a transaction is exitable.
+There may be further ways to optimize these two options.
+
+We still need to provide a deterministic ordering of exits by some priority.
+MoreVP exits are given a priority based on the position in the Plasma chain of the most recently included (youngest) input to that transaction.
+Unlike the MVP protocol, we give each input and output to a transaction the same priority.
+This should be implemented by inserting a single “exit” object into a priority queue of exits and tracking a list of inputs or outputs to be paid out once the exit is processed.
+
+
+#### Proving Canonicity
+
+Whether unspent inputs or unspent outputs are paid out in an exit depends on the canonicity of the referenced transaction.
```

Comment:
add "... and is fully independent of the piggybackings that happen" or similar, let's explicitly steer clear of this old confusion